src/CLI/usbguard-allow-device.cpp

```diff
@@ -75,14 +76,47 @@ namespace usbguard
     argc -= optind;
     argv += optind;
 
-    if (argc != 1) {
-      showHelp(std::cerr);
-      return EXIT_FAILURE;
+    usbguard::IPCClient ipc(/*connected=*/true);
+
+    if (argc == 1) {
+      id = std::stoul(argv[0]);
+      ipc.applyDevicePolicy(id, Rule::Target::Allow, permanent);
+    }
+    else {
+      /*
+       * Create a rulestring from the arguments
+       */
+      std::string rule_string = argv[0];
+
+      for (int i = 1; i < argc; i++) {
+        rule_string.append(" ").append(argv[i]);
+      }
+
+      usbguard::Rule rule;
+      try {
+        rule = Rule::fromString(rule_string);
+      }
+      catch (...) {
+        showHelp(std::cerr);
+        return EXIT_FAILURE;
+      }
+
+      /*
+       * Create a ruleset which will contain the requested rule
+      */
+      auto my_rule_set = std::dynamic_pointer_cast<RuleSet>(std::make_shared<MemoryRuleSet>(nullptr));
```

Comment:
I don't think you have to create a RuleSet here. Wouldn't it be possible to use the [Rule::appliesTo](https://github.com/USBGuard/usbguard/blob/27801e8800b6c2ebf34a6104d318ae541ff66706/src/Library/public/usbguard/Rule.hpp#L490) method? i.e. something like:
```
if (rule.appliesTo(rule_device)) ... 
```

Reply:
Yes, you are right. It is a lot more elegant solution.